### DISCUSSION TITLE
Clarification of the 'on-mesh' prefix flag role

### DISCUSSION THREAD

**Discussion by kderda** [openthread/openthread#9742] [Q&A] [answered]
Hello, 
Even though I work with the Thread for some time I have always wondered what's the exact meaning of the 'on-mesh' flag of prefix configuration. I was looking for the answer both in the Specification and OT's source code but I could not find any description other than similar to "on-mesh flag indicates that prefix is on-mesh". 

What are the implications of configuring a prefix without this flag? I know that it means that a given prefix is not a valid OMR prefix but that must be more and I am very curious :smile:.

**Comment by jwhui** [comment, marked as answer]
On-mesh prefix indicates whether or not to perform Address Query to discover destination. It is analogous to the "on-link" flag in Prefix Information Option contained within IPv6 Router Advertisements.
